scripts/redux/actions.js

```diff
@@ -258,29 +258,44 @@ const blogActions = {
 };
 
 const speakersActions = {
-  fetchList: () => (dispatch) => {
+  fetchList: () => (dispatch, getState) => {
     dispatch({
       type: FETCH_SPEAKERS,
     });
 
-    firebase.firestore()
-      .collection('speakers')
-      .orderBy('order', 'asc')
-      .get()
-      .then((snaps) => {
-        const list = snaps.docs
-          .map((snap) => Object.assign({}, snap.data(), { id: snap.id }));
+    const state = getState();
+    const sessionsPromise = Object.keys(state.sessions.list).length
+      ? Promise.resolve(state.sessions)
+      : sessionsActions.fetchList()(dispatch, getState);
 
-        const obj = list.reduce(
-          (acc, curr) => Object.assign({}, acc, { [curr.id]: curr }),
-          {}
-        );
+    const speakersPromise = new Promise((resolve, reject) => {
+      firebase.firestore()
+        .collection('speakers')
+        .orderBy('order', 'asc')
+        .get()
+        .then((snaps) => {
+          resolve(snaps.docs.map((snap) => Object.assign({}, snap.data(), { id: snap.id })));
+        })
+        .catch(reject);
+    });
 
+    return Promise.all([speakersPromise, sessionsPromise])
+      .then(([speakers, sessions]) => {
+        const updatedSpeakersList = speakers.map((speaker) => Object.assign({}, speaker, {
+          tags: sessions.objBySpeaker[speaker.id]
+            ? sessions.objBySpeaker[speaker.id].reduce((aggr, session) => {
+              return aggr.concat(session.tags
+                ? session.tags.filter((tag) => aggr.indexOf(tag) < 0)
```

Comment:
indexOf < 0 ? So you select which does not include specified tags?

Reply:
Is it possible to break down this filtering into smaller pieces? To make it more readable

Reply:
I select unique values to the array.
Will try to split